### ISSUE TITLE
Basic Auth support

### ISSUE THREAD

**Issue by v4rakh** [ngosang/restic-exporter#3]
Hi, thanks for providing this. Would it be possible to have basic auth support in and being able to set user and pass via env var?

**Comment by ngosang** [comment]
You mean in the `/metrics` endpoint?

**Comment by v4rakh** [comment]
Yes, exactly.

**Comment by ngosang** [comment]
The purpose of this exported is to be consumed by Prometheus. I never saw an exporter with basic auth and I think the prometheus client does not support it.
If you are worried about unauthorized access to the endpoint, you can do it with network policies. If you are using Docker containers it's even easier. Anyway there is no much to be worried about, the info exposed is not sensitive.

Btw, new release is out => https://github.com/ngosang/restic-exporter/releases/tag/1.1.0

**Comment by v4rakh** [comment]
@ngosang Maybe we're talking about different things, not sure, but the `/metrics` endpoint which is being scraped and the scraping being configured in the `prometheus.yml` allows to specify basic auth. A lot of the exporters allow to secure the exposed metrics endpoint, like the default node-exporter or the node-systemd-exporter.

**Comment by ngosang** [comment]
You are right. It's supported by the python client library and Prometheus. I didn't know about that.
I'm not interested in implement this feature but I can accept PRs.
Example code => https://github.com/prometheus/client_python#handlers-for-authentication
